### PR TITLE
Benchmark

### PR DESCRIPTION
**Pull request by alextwoods**
*Description of changes:*
Follow approach for benchmarking from v3 sdk.

This also adds generation of the `VERSION` file - following the practice we've done in other gems (including the new Hearth). 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  id-token: write # required for OIDC
+  contents: read
+
+jobs:
+  benchmark:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [3.2]
```
Comment (mullermp) [inline .github/workflows/benchmark.yml:22]: We may as well run this on all 3.0+ based ruby versions

File: benchmark/gems/hearth.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Benchmark
+  module Gems
+    class Core < Benchmark::Gem
```
Comment (mullermp) [inline benchmark/gems/hearth.rb:5]: I feel like hearth benchmark is pretty useless. It's also not aws/core ..

File: benchmark/metrics.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Benchmark
+  module Metrics
+    # put metrics generated from `run_benchmarks` to cloudwatch
+    def self.put_metric(client, dims, timestamp, k, v)
```
Comment (mullermp) [inline benchmark/metrics.rb:6]: Positional args for this is kind of weird .. I know it's like this in v3

**Review by mullermp — APPROVED** [review]

File: benchmark/gems/hearth.rb (hunk 2 above)
Comment (alextwoods) [reply benchmark/gems/hearth.rb]: Updated the name to be Hearth instead. I do think its useful to track the gem size and require time over time.

File: benchmark/metrics.rb (hunk 3 above)
Comment (alextwoods) [reply benchmark/metrics.rb]: Updated to kwargs

File: .github/workflows/benchmark.yml (hunk 1 above)
Comment (alextwoods) [reply .github/workflows/benchmark.yml]: Added

**Pull request merged**